dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,142 @@
+- pentagon:
+    - an H3 cell which is a **topological** pentagon
+- `DirectedEdge`
+    - an H3 index of `mode 2`
+    - representing a traversal from an origin cell to an adjacent destination cell
+- grid:
+    - the graph with nodes corresponding to H3 cells, and edges given by pairs of adjacent cells
+    - for example, `gridDistance` is the minimal number of edges in a graph path connecting two cells
+
+### Notes
+
+We realize that "hexagon" will still be used informally to refer to the concept of "cell" (As the development team, we do it ourselves!).
+This should be expected in casual, informal discussions of H3.
+However, when **precision** is required, we advise the use of strict technical terms like "index", "cell", "hexagon", "pentagon", etc.
+In the codebase and in the documentation, strictly correct terminology should **always** be used.
+
+
+## Use of "Topological"
+
+We should clarify in the documentation that H3 works with **topological** hexagons and pentagons, rather than true **geometric** hexagons and pentagons.
+
+See, for example: https://github.com/uber/h3-js/issues/53
+
+
+## General Function Names
+
+|          Current name         |     Proposed name     |
+|-------------------------------|-----------------------|
+| *Does Not Exist (DNE)*        | `isValidIndex`        |
+| `h3IsValid`                   | `isValidCell`         |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge` |
+| `h3IsPentagon`                | `isPentagon`          |
+| `h3ToParent`                  | `cellParent`          |
+| `h3ToChildren`                | `cellChildren`        |
+| `h3IndexesAreNeighbors`       | `cellsAreNeighbors`   |
+| `geoToH3`                     | `geoToCell`           |
+| `h3ToGeo`                     | `cellToGeo`           |
+| `numHexagons`                 | `numCells`            |
+| `getRes0Indexes`              | `getRes0Cells`        |
+| `getPentagonIndexes`          | `getPentagons`        |
+| `h3GetBaseCell`               | `getBaseCellNumber`   |
+
+### naming note
+
+- `isValid*` should mean that a full validity check is made
+- without `Valid` (like in the case of `isPentagon`), we do not guarantee
+  that a full validity check is made; instead, a user should assume only a
+  minimal bit mask check is done
+    - we could imagine a `isValidPengaton` function, if full verification
+      would be convenient
+- similarly, a function like `cellsAreNeighbors` will assume each cell
+  has passed the `isValidCell` check; the function will do only minimal
+  work to determine if they are neighbors
+
+### todo
+- Do we have general guidance on when to use a `get` prefix with a function and when not?
+
+
+## H3 Grid Functions
+
+| Current name |     Proposed name      |
+|--------------|------------------------|
+| `h3Distance` | `gridDistance`         |
+| `h3Line`     | `gridPathCell`         |
+| *DNE*        | `gridPathEdge`         |
+| *DNE*        | `gridPathDirectedEdge` |
```

Comment:
It is a little difficult for me to mentally parse the name `gridPathCell`. Perhaps `cellPath` (as previously discussed), or maybe make the last word plural (`gridPathCells`, `gridPathEdges`, ...), would that better communicate it is a path made up of cells?

Reply:
I like pluralizing.